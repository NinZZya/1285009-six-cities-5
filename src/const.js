export const AppPath = {
  ROOT: `/`,
  LOGIN: `/login`,
  CITY: `/city`,
  OFFER: `/offer`,
  FAVORITES: `/favorites`,
  NOT_FOUND: `/not-found`,
};

export const CITIES = {
  1: {id: 1, name: `Paris`},
  2: {id: 2, name: `Cologne`},
  3: {id: 3, name: `Brussels`},
  4: {id: 4, name: `Amsterdam`},
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};

export const SortType = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const STARS = [
  {value: 5, title: `perfect`},
  {value: 4, title: `good`},
  {value: 3, title: `not bad`},
  {value: 2, title: `badly`},
  {value: 1, title: `terribly`},
];
