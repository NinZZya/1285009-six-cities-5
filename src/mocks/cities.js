export default {
  1: {
    id: 1,
    name: `Paris`,
    coords: [48.864716, 2.349014],
  },
  2: {
    id: 2,
    name: `Cologne`,
    coords: [50.9415491, 6.9408247],
  },
  3: {
    id: 3,
    name:
    `Brussels`,
    coords: [50.8503396, 4.3517103],
  },
  4: {
    id: 4,
    name: `Amsterdam`,
    coords: [52.38333, 4.9],
  },
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};
