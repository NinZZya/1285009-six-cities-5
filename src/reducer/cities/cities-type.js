export const SETE_CITIES = `SETE_CITIES`;
export const CHANGE_ACTIVE_CITY_ID = `CHANGE_ACTIVE_CITY_ID`;
