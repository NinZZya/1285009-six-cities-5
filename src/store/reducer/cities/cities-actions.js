import * as CitiesType from './cities-types';
import {makeActionCreator} from '../../../utils/utils';


export const setCities = makeActionCreator(CitiesType.SET_CITIES);
export const changeActiveCityId = makeActionCreator(CitiesType.CHANGE_ACTIVE_CITY_ID);
