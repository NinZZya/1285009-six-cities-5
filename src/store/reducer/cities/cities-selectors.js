import NameSpace from '~/store/reducer/name-space';


const NAME_SPACE = NameSpace.CITIES;


export const getActiveCityId = (state) => state[NAME_SPACE].activeId;
export const getCities = (state) => state[NAME_SPACE].cities;
