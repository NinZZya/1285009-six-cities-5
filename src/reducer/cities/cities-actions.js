import * as CitiesType from './cities-type';
import {makeActionCreator} from '../../utils/utils';


export const updateCities = makeActionCreator(CitiesType.UPDATE_CITIES);
export const changeActiveCityId = makeActionCreator(CitiesType.CHANGE_ACTIVE_CITY_ID);
