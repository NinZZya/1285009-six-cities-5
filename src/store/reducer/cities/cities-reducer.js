import * as CitiesType from './cities-type';
import {extend} from '../../../utils/utils';
import {DEFAULT_CITY_ID} from '../../../constants/const';


const cities = {
  1: {id: 1, name: `Paris`},
  2: {id: 2, name: `Cologne`},
  3: {id: 3, name: `Brussels`},
  4: {id: 4, name: `Amsterdam`},
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};

const initialState = {
  cities,
  activeId: DEFAULT_CITY_ID,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case CitiesType.SETE_CITIES:
      return extend(state, {
        cities: action.payload,
      });
    case CitiesType.CHANGE_ACTIVE_CITY_ID:
      return extend(state, {
        activeId: action.payload,
      });
    default:
      return state;
  }
};
