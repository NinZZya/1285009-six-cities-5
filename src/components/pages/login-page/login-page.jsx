import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import PageContainer from '@components/page-container/page-container';
import Container from '@components/container/container';
import LoginForm from '@components/login-form/login-form';
import {authAsync} from '@reducer/user/user-operations';
import * as UserAction from '@reducer/user/user-actions';
import * as UserSelector from '@reducer/user/user-selectors';
import * as CitiesSelector from '@reducer/cities/cities-selectors';
import * as Type from '@/constants/types';
import {AppPath, UserStatus} from '@/constants/const';


const ContainerType = {
  PAGE: `login`,
  LOGIN: `page__login`,
};

const LoginPage = (props) => {
  const {
    activeCityId,
    userStatus,
    error,
    cities,
    auth,
    changeUserStatus,
  } = props;

  const city = cities[activeCityId];
  const cityPath = `${AppPath.CITY}/${activeCityId}`;

  return (
    <PageContainer type={ContainerType.PAGE}>
      <Container type={ContainerType.LOGIN}>
        <section className="login">
          <h1 className="login__title">Sign in</h1>
          <LoginForm
            onAuth={(authData) => {
              changeUserStatus(UserStatus.RESPONSE);
              auth(authData);
            }}
            error={error}
            userStatus={userStatus}
          />
        </section>
        <section className="locations locations--login locations--current">
          <div className="locations__item">
            <Link to={cityPath} className="locations__item-link">
              <span>{city.name}</span>
            </Link>
          </div>
        </section>
      </Container>
    </PageContainer>
  );
};

LoginPage.propTypes = {
  activeCityId: Type.ID,
  userStatus: Type.USER_STATUS,
  error: Type.USER_ERROR,
  cities: Type.CITIES,
  auth: Type.FUNCTION,
  changeUserStatus: Type.FUNCTION,
};

const mapStateToProps = (state) => ({
  activeCityId: CitiesSelector.getActiveCityId(state),
  cities: CitiesSelector.getCities(state),
  userStatus: UserSelector.getUserStatus(state),
  error: UserSelector.getError(state),
});

const mapDispatchToProps = (dispatch) => ({
  auth: (authData) => {
    dispatch(authAsync(authData));
  },
  changeUserStatus: (userStatus) => {
    dispatch(UserAction.changeUserStatus(userStatus));
  },
});


export {LoginPage};
export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
