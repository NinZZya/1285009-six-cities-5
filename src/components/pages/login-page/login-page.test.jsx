import React from 'react';
import {Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {LoginPage} from './login-page';
import history from '../../../constants/history';
import {UserStatus} from '../../../constants/const';
import cities from '../../../mocks/cities';


const ACTIVE_CITY_ID = 4;

const ERROR = `Error`;
const testing = () => {};

describe(`Should LoginPage render correctly`, () => {
  it(`Should LoginPage render correctly no auth`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <LoginPage
              activeCityId={ACTIVE_CITY_ID}
              userStatus={UserStatus.NO_AUTH}
              cities={cities}
              auth={testing}
              changeUserStatus={testing}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should LoginPage render correctly error`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <LoginPage
              activeCityId={ACTIVE_CITY_ID}
              userStatus={UserStatus.AUTH_ERROR}
              error={ERROR}
              cities={cities}
              auth={testing}
              changeUserStatus={testing}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should LoginPage render correctly responce`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <LoginPage
              activeCityId={ACTIVE_CITY_ID}
              userStatus={UserStatus.AUTH}
              cities={cities}
              auth={testing}
              changeUserStatus={testing}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
