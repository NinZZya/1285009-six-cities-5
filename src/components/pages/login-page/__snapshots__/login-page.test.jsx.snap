// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should LoginPage render correctly Should LoginPage render correctly error 1`] = `
<main
  className="page__main page__main--login "
>
  <div
    className="container page__login-container "
  >
    <section
      className="login"
    >
      <h1
        className="login__title"
      >
        Sign in
      </h1>
      <form
        action="#"
        className="login__form form"
        method="post"
        onSubmit={[Function]}
      >
        <div
          className="login__input-wrapper form__input-wrapper"
        >
          <label
            className="visually-hidden"
          >
            E-mail
          </label>
          <input
            className="login__input form__input"
            disabled={false}
            name="email"
            placeholder="Email"
            required=""
            type="email"
          />
        </div>
        <div
          className="login__input-wrapper form__input-wrapper"
        >
          <label
            className="visually-hidden"
          >
            Password
          </label>
          <input
            className="login__input form__input"
            disabled={false}
            name="password"
            placeholder="Password"
            required=""
            type="password"
          />
        </div>
        <button
          className="login__submit form__submit button"
          disabled={false}
          type="submit"
        >
          Sign in
        </button>
        <p
          style={
            Object {
              "color": "#ff0000",
              "fontSize": "12px",
            }
          }
        >
          Error
        </p>
      </form>
    </section>
    <section
      className="locations locations--login locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="/city/4"
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </section>
  </div>
</main>
`;

exports[`Should LoginPage render correctly Should LoginPage render correctly no auth 1`] = `
<main
  className="page__main page__main--login "
>
  <div
    className="container page__login-container "
  >
    <section
      className="login"
    >
      <h1
        className="login__title"
      >
        Sign in
      </h1>
      <form
        action="#"
        className="login__form form"
        method="post"
        onSubmit={[Function]}
      >
        <div
          className="login__input-wrapper form__input-wrapper"
        >
          <label
            className="visually-hidden"
          >
            E-mail
          </label>
          <input
            className="login__input form__input"
            disabled={false}
            name="email"
            placeholder="Email"
            required=""
            type="email"
          />
        </div>
        <div
          className="login__input-wrapper form__input-wrapper"
        >
          <label
            className="visually-hidden"
          >
            Password
          </label>
          <input
            className="login__input form__input"
            disabled={false}
            name="password"
            placeholder="Password"
            required=""
            type="password"
          />
        </div>
        <button
          className="login__submit form__submit button"
          disabled={false}
          type="submit"
        >
          Sign in
        </button>
      </form>
    </section>
    <section
      className="locations locations--login locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="/city/4"
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </section>
  </div>
</main>
`;

exports[`Should LoginPage render correctly Should LoginPage render correctly responce 1`] = `
<main
  className="page__main page__main--login "
>
  <div
    className="container page__login-container "
  >
    <section
      className="login"
    >
      <h1
        className="login__title"
      >
        Sign in
      </h1>
      <form
        action="#"
        className="login__form form"
        method="post"
        onSubmit={[Function]}
      >
        <div
          className="login__input-wrapper form__input-wrapper"
        >
          <label
            className="visually-hidden"
          >
            E-mail
          </label>
          <input
            className="login__input form__input"
            disabled={false}
            name="email"
            placeholder="Email"
            required=""
            type="email"
          />
        </div>
        <div
          className="login__input-wrapper form__input-wrapper"
        >
          <label
            className="visually-hidden"
          >
            Password
          </label>
          <input
            className="login__input form__input"
            disabled={false}
            name="password"
            placeholder="Password"
            required=""
            type="password"
          />
        </div>
        <button
          className="login__submit form__submit button"
          disabled={false}
          type="submit"
        >
          Sign in
        </button>
      </form>
    </section>
    <section
      className="locations locations--login locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="/city/4"
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </section>
  </div>
</main>
`;
